mne/time_frequency/tfr.py

```diff
@@ -878,15 +878,32 @@ def crop(self, tmin=None, tmax=None):
             Start time of selection in seconds.
         tmax : float | None
             End time of selection in seconds.
+        fmin : float | None
+            Lowest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
+        fmax : float | None
+            Highest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
 
         Returns
         -------
         inst : instance of AverageTFR
             The modified instance.
         """
-        mask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
-        self.times = self.times[mask]
-        self.data = self.data[..., mask]
+        time_mask = _time_mask(self.times, tmin, tmax,
+                               sfreq=self.info['sfreq'])
+
+        self.times = self.times[time_mask]
+        self.data = self.data[..., time_mask]
+
+        freq_mask = _freq_mask(self.freqs, sfreq=self.info['sfreq'],
+                               fmin=fmin, fmax=fmax)
+
+        self.freqs = self.freqs[freq_mask]
+        self.data = self.data[..., freq_mask, :]
```

Comment:
freq_mask is always a bool here. So this code will make 3 copies of the data in memory. You should avoid the masking of time or freq is not requested.